Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -628,270 +972,182 @@ bool UserCmd_Process(uint client, const wstring &wscCmd)
-
-		string path = "./flhook_logs/logactivity " + string(buffer) + ".log";
-
-		// Lines list to add in new file.
-		vector<string> Lines;
-
-		Lines.push_back("DOCK: mobiledockClients | Size = " + to_string(mobiledockClients.size()));
-
-		vector<wstring> DOCKcharnames;
-		for (map<uint, CLIENT_DATA>::iterator it = mobiledockClients.begin(); it != mobiledockClients.end(); ++it)
+		idToCarrierInfoMap[iClientID] = &nameToCarrierInfoMap[charname];
+		idToCarrierInfoMap[iClientID]->lastCarrierLogin = time(nullptr);
+		for (const auto& dockedShipName : idToCarrierInfoMap[iClientID]->dockedShipList)
 		{
-			wstring charname;
-
-			try
-			{
-				charname = (const wchar_t*)Players.GetActiveCharacterName(it->first);
-			}
-			catch (...)
-			{
-				charname = L"<Error>";
-			}
-
-			string ID = to_string(it->first);
-			string Charname = wstos(charname);
-			string Type = it->second.iDockingModulesInstalled == 0 ? "Docked" : "Carrier";
-			if (Type == "Carrier")
-			{
-				wstring docked = L"";
-				for (map<wstring, wstring>::iterator cit = it->second.mapDockedShips.begin(); cit != it->second.mapDockedShips.end(); cit++)
-				{
-					if (docked != L"")
-						docked += L" | ";
-					docked += cit->first;
-				}
-
-				Type += "[" + to_string(it->second.iDockingModulesInstalled) + "](" + wstos(docked) + ")";
-			}
-			else
-			{
-
-				if (it->second.wscDockedWithCharname.empty())
-				{
-					if (!it->second.mobileDocked)
-						continue;
-					Type += "[" + wstos(L"<Error>") + "]";
-				}
-				else
-					Type += "[" + wstos(it->second.wscDockedWithCharname) + "]";
-			}
-			string State = "";
-
-			if (it->first == 0 || it->first > MAX_CLIENT_ID)
-				State += "Out of range";
-
-			if (find(DOCKcharnames.begin(), DOCKcharnames.end(), charname) != DOCKcharnames.end())
+			uint dockedClientID = HkGetClientIdFromCharname(dockedShipName.c_str());
+			if (dockedClientID != -1)
 			{
-				if (State != "")
-					State += " | ";
-				State += "Doubled";
+				PrintUserCmdText(dockedClientID, L"Carrier ship has come online.");
 			}
-
-			if (State == "")
-				State = "Fine";
-
-			if (charname != L"<Error>")
-				DOCKcharnames.push_back(charname);
-
-			Lines.push_back(ID + " " + Charname + " " + Type + " " + State);
 		}
-
-		Lines.push_back("");
-
-
-		vector<string> SERVERlines;
-
-		vector<wstring> SERVERcharnames;
-		vector<uint> IDs;
-		struct PlayerData *pPD = 0;
-		while (pPD = Players.traverse_active(pPD))
+	}
+	else if (nameToDockedInfoMap.count(charname))
+	{
+		idToDockedInfoMap[iClientID] = &nameToDockedInfoMap[charname];
+		uint carrierClientID = HkGetClientIdFromCharname(idToDockedInfoMap[iClientID]->carrierName.c_str());
+		if (carrierClientID != -1)
 		{
-			uint clientID;
-			try
+			const auto& shipInfo = Archetype::GetShip(Players[iClientID].iShipArchetype);
+			wstring& shipName = HkGetWStringFromIDS(shipInfo->iIdsName);
+			PrintUserCmdText(carrierClientID, L"INFO: Docked %ls, %ls is standing by for launch", shipName.c_str(), charname);

```

Comment:
Instead of saying docked this should talk about the docked ship having come online